### PR TITLE
Fix #10979. GetFlaggedPostsForChannel is sync now

### LINKED ISSUE #10979: Migrate "Post.GetFlaggedPostsForChannel" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15849)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetFlaggedPostsForChannel` in the `Post` store.

The expected way to implement it is, go to the `Post` store implementation in the `store/sqlstore/` directory, modify the method `GetFlaggedPostsForChannel` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by rvillablanca**
I have a doubt, the method is called `GetFlaggedPostsForChannel` but it doesn't return a channel anymore so, should it be renamed ?

### CODE REVIEW

**Comment by hanzei** [comment]
Will review next week. Sorry for the delay.

File: store/sqlstore/post_store.go [hunk 1]
```diff
@@ -236,33 +236,31 @@ func (s *SqlPostStore) GetFlaggedPostsForTeam(userId, teamId string, offset int,
...
+		WHERE
+			Id IN (SELECT Name FROM Preferences WHERE UserId = :UserId AND Category = :Category)
+			AND ChannelId = :ChannelId
+			AND DeleteAt = 0
+		ORDER BY CreateAt DESC
+		LIMIT :Limit OFFSET :Offset`
+
+	if _, err := s.GetReplica().Select(&posts, query, map[string]interface{}{"UserId": userId, "Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "ChannelId": channelId, "Offset": offset, "Limit": limit}); err != nil {
+		return nil, model.NewAppError("SqlPostStore.GetFlaggedPostsForChannel", "store.sql_post.get_flagged_posts.app_error", nil, err.Error(), http.StatusInternalServerError)
+	} else {
```
Comment (jespino) [inline store/sqlstore/post_store.go]: You can remove this else here.

**Comment by jespino** [comment]
Hi @rvillablanca, the name of the function is correct, it gets the flagged posts for a channel, that means, you get a list of posts that are flagged for the user that you pass as parameter, in the channel that you pass as parameter.

File: store/sqlstore/post_store.go (hunk 1 above)
Comment (rvillablanca) [reply store/sqlstore/post_store.go]: Fixed!

**Comment by hanzei** [comment]
Thanks @rvillablanca :rocket:

**Pull request merged**